src/DIRAC/WorkloadManagementSystem/Agent/JobCleaningAgent.py

```diff
@@ -230,11 +232,11 @@ def _deleteRemoveJobs(self, jobList, remove=False):
             if not res["OK"]:
                 self.log.error("No DN found", f"for {user}")
                 return res
-            wmsClient = WMSClient(useCertificates=True, delegatedDN=res["Value"][0], delegatedGroup=ownerGroup)
             if remove:
+                wmsClient = WMSClient(useCertificates=True, delegatedDN=res["Value"][0], delegatedGroup=ownerGroup)
```

Comment:
Don't you need to move the logic of `removeJob` out of the `JobManagerHandler` too?